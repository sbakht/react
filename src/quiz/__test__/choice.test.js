import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Choice from '../choice';
import toJSON from 'enzyme-to-json';

test('renders default', () => {
  var choice = {};
  const component = shallow(
    <Choice choice={choice} />
  );

  expect(toJSON(component)).toMatchSnapshot();
  expect(component.hasClass('choice')).toBe(true);
});

test('renders selected choice', () => {
  var choice = {};
  const component = shallow(
    <Choice choice={choice} isSelected={true} />
  );

  expect(toJSON(component)).toMatchSnapshot();
  expect(component.hasClass('selected')).toBe(true);
});

test('choice text', () => {
  var choice = { text : 'choice#1'};
  const component = shallow(
    <Choice choice={choice} />
  );

  expect(toJSON(component)).toMatchSnapshot();
  expect(component.text()).toBe("choice#1");
});

test('calls parent on choice select with choice', () => {
  var fn = jest.fn();
  var choice = { text : 'choice#1'};
  const component = shallow(
    <Choice choice={choice} onClick={fn} />
  );

  component.find('input').simulate('click');
  expect(fn.mock.calls.length).toBe(1);
  expect(fn.mock.calls[0][0]).toEqual(choice);
});

test('has proptypes', () => {
  expect(Choice.propTypes).toBeDefined();
  
  Object.keys(Choice.propTypes).map((key) => {
    expect(Choice.propTypes[key]).toBeDefined();
  });
});