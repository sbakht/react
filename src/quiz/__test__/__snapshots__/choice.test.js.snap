// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`choice text 1`] = `
<label
  className="choice"
>
  <input
    name="choices"
    onClick={[Function]}
    type="radio"
  />
  choice#1
</label>
`;

exports[`renders default 1`] = `
<label
  className="choice"
>
  <input
    name="choices"
    onClick={[Function]}
    type="radio"
  />
</label>
`;

exports[`renders selected choice 1`] = `
<label
  className="choice selected"
>
  <input
    checked={true}
    name="choices"
    onClick={[Function]}
    type="radio"
  />
</label>
`;
